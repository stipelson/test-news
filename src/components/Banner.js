import React from 'react';
import PropTypes from 'prop-types';
import Row from 'emerald-ui/lib/Row';
import Col from 'emerald-ui/lib/Col';
import Button from 'emerald-ui/lib/Button';

import ParallaxImage from '../assets/images/parallax-2x.jpg';

const Banner = ({
  title,
  content,
  buttonLabel,
  buttonAriaLabel,
  buttonAction,
}) => {
  return (
    <section id="banner" className="banner">
      <img className="banner-background" src={ParallaxImage} />
      <div className="container banner-body">
        <Row>
          <Col xs={6} className="sample-col">
            <h2 className="text-white banner-title">{title}</h2>
            <p className="text-white">{content}</p>
            <Button
              className="eui-btn-inverted btn-padding-lg"
              shape="outline"
              size="sm"
              ariaLabel={buttonAriaLabel}
              onClick={buttonAction}
            >
              <span>{buttonLabel}</span>
            </Button>
          </Col>
        </Row>
      </div>
    </section>
  );
};

Banner.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  buttonLabel: PropTypes.string,
  buttonAriaLabel: PropTypes.string,
  buttonAction: PropTypes.func,
};

Banner.defaultProps = {
  title: '',
  content: '',
  buttonLabel: '',
  buttonAriaLabel: '',
  buttonAction: () => {
    return false;
  },
};

export default Banner;
