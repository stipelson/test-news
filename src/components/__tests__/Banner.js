import React from 'react';
import renderer from 'react-test-renderer';
import Col from 'emerald-ui/lib/Col';
import Button from 'emerald-ui/lib/Button';

import ParallaxImage from '../../assets/images/parallax-2x.jpg';
import Banner from '../Banner';

describe('Banner: render', () => {
  it('renders banner correctly', () => {
    const tree = renderer.create(<Banner />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has the section class', () => {
    const component = renderer.create(<Banner />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const section = instance.findByType('section');
    expect(section.props.className).toEqual('banner');
  });

  it('img has the class', () => {
    const component = renderer.create(<Banner />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const img = instance.findByType('img');
    expect(img.props.className).toEqual('banner-background');
  });

  it('container has the class', () => {
    const component = renderer.create(<Banner />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    // Verify container class
    const section = instance.findByType('section');
    const container = section.findByType('div');
    expect(container.props.className).toEqual('container banner-body');

    // Verify col class
    const col = section.findByType(Col);
    expect(col.props.className).toEqual('sample-col');

    // Verify title class
    const title = section.findByType('h2');
    expect(title.props.className).toEqual('text-white banner-title');

    // Verify body class
    const body = section.findByType('p');
    expect(body.props.className).toEqual('text-white');
  });

  it('Button has the class and caption', () => {
    const component = renderer.create(<Banner />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    // Verify button class
    const button = instance.findByType(Button);
    expect(button.props.className).toEqual('eui-btn-inverted btn-padding-lg');
    expect(button.props.shape).toEqual('outline');

    // Verify button > span exist
    const spans = button.findAllByProps({ className: 'caption' });
    expect(spans.length).toEqual(1);
  });
});

describe('Banner: content', () => {
  it('Banner render props content', () => {
    const actionMock = jest.fn();
    const component = renderer.create(
      <Banner
        title="Test title"
        content="Test content"
        buttonLabel="Test button label"
        buttonAriaLabel="Test button aria label"
        buttonAction={actionMock}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    const img = instance.findByType('img');
    expect(img.props.src).toEqual(ParallaxImage);

    // Verify title content
    const title = instance.findByType('h2');
    expect(title.props.children).toEqual('Test title');

    // Verify body content
    const body = instance.findByType('p');
    expect(body.props.children).toEqual('Test content');

    // Verify Button props render
    const button = instance.findByType(Button);
    expect(button.props.ariaLabel).toEqual('Test button aria label');

    const span = button.findByProps({ className: 'caption' });
    expect(span.children[0].children).toEqual(['Test button label']);
  });
});
