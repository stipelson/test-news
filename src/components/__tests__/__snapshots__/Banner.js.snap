// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner: content Banner render props content 1`] = `
<section
  className="banner"
  id="banner"
>
  <img
    className="banner-background"
    src="test-file-stub"
  />
  <div
    className="container banner-body"
  >
    <div
      className="eui-row"
    >
      <div
        className="eui-col sample-col eui-col-xs-6"
      >
        <h2
          className="text-white banner-title"
        >
          Test title
        </h2>
        <p
          className="text-white"
        >
          Test content
        </p>
        <button
          aria-label="Test button aria label"
          className="eui-btn eui-btn-default eui-btn-outline eui-btn-sm eui-btn-inverted btn-padding-lg"
          onClick={[MockFunction]}
          role="button"
          type="button"
        >
          <div
            className="eui-btn-content"
          >
            <span
              className="caption"
            >
              <span>
                Test button label
              </span>
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`Banner: render Button has the class and caption 1`] = `
<section
  className="banner"
  id="banner"
>
  <img
    className="banner-background"
    src="test-file-stub"
  />
  <div
    className="container banner-body"
  >
    <div
      className="eui-row"
    >
      <div
        className="eui-col sample-col eui-col-xs-6"
      >
        <h2
          className="text-white banner-title"
        >
          
        </h2>
        <p
          className="text-white"
        >
          
        </p>
        <button
          aria-label=""
          className="eui-btn eui-btn-default eui-btn-outline eui-btn-sm eui-btn-inverted btn-padding-lg"
          onClick={[Function]}
          role="button"
          type="button"
        >
          <div
            className="eui-btn-content"
          >
            <span
              className="caption"
            >
              <span>
                
              </span>
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`Banner: render container has the class 1`] = `
<section
  className="banner"
  id="banner"
>
  <img
    className="banner-background"
    src="test-file-stub"
  />
  <div
    className="container banner-body"
  >
    <div
      className="eui-row"
    >
      <div
        className="eui-col sample-col eui-col-xs-6"
      >
        <h2
          className="text-white banner-title"
        >
          
        </h2>
        <p
          className="text-white"
        >
          
        </p>
        <button
          aria-label=""
          className="eui-btn eui-btn-default eui-btn-outline eui-btn-sm eui-btn-inverted btn-padding-lg"
          onClick={[Function]}
          role="button"
          type="button"
        >
          <div
            className="eui-btn-content"
          >
            <span
              className="caption"
            >
              <span>
                
              </span>
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`Banner: render has the section class 1`] = `
<section
  className="banner"
  id="banner"
>
  <img
    className="banner-background"
    src="test-file-stub"
  />
  <div
    className="container banner-body"
  >
    <div
      className="eui-row"
    >
      <div
        className="eui-col sample-col eui-col-xs-6"
      >
        <h2
          className="text-white banner-title"
        >
          
        </h2>
        <p
          className="text-white"
        >
          
        </p>
        <button
          aria-label=""
          className="eui-btn eui-btn-default eui-btn-outline eui-btn-sm eui-btn-inverted btn-padding-lg"
          onClick={[Function]}
          role="button"
          type="button"
        >
          <div
            className="eui-btn-content"
          >
            <span
              className="caption"
            >
              <span>
                
              </span>
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`Banner: render img has the class 1`] = `
<section
  className="banner"
  id="banner"
>
  <img
    className="banner-background"
    src="test-file-stub"
  />
  <div
    className="container banner-body"
  >
    <div
      className="eui-row"
    >
      <div
        className="eui-col sample-col eui-col-xs-6"
      >
        <h2
          className="text-white banner-title"
        >
          
        </h2>
        <p
          className="text-white"
        >
          
        </p>
        <button
          aria-label=""
          className="eui-btn eui-btn-default eui-btn-outline eui-btn-sm eui-btn-inverted btn-padding-lg"
          onClick={[Function]}
          role="button"
          type="button"
        >
          <div
            className="eui-btn-content"
          >
            <span
              className="caption"
            >
              <span>
                
              </span>
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`Banner: render renders banner correctly 1`] = `
<section
  className="banner"
  id="banner"
>
  <img
    className="banner-background"
    src="test-file-stub"
  />
  <div
    className="container banner-body"
  >
    <div
      className="eui-row"
    >
      <div
        className="eui-col sample-col eui-col-xs-6"
      >
        <h2
          className="text-white banner-title"
        >
          
        </h2>
        <p
          className="text-white"
        >
          
        </p>
        <button
          aria-label=""
          className="eui-btn eui-btn-default eui-btn-outline eui-btn-sm eui-btn-inverted btn-padding-lg"
          onClick={[Function]}
          role="button"
          type="button"
        >
          <div
            className="eui-btn-content"
          >
            <span
              className="caption"
            >
              <span>
                
              </span>
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</section>
`;
